kf_lib_data_ingest/etl/extract/extract.py

```diff
@@ -162,6 +163,27 @@ def _clean_up_df(self, df):
             lambda x: numeric_to_str(x, replace_na=True, na=None)
         )
 
+    def _obvert_visibility(self, df):
+        """
+        If something is visible, then also record that it's not hidden, and
+        vice versa.
+        """
+
+        def flip_col(in_col):
+            out_col = in_col.copy()
+            out_col[out_col.notnull()] = (
+                ~in_col[in_col.notnull()].astype(bool)
+            ).astype(object)
+            return out_col
+
+        for concept in concept_set:
+            if concept.HIDDEN in df and concept.VISIBLE not in df:
+                df[concept.VISIBLE] = flip_col(df[concept.HIDDEN])
+            elif concept.VISIBLE in df and concept.HIDDEN not in df:
+                df[concept.HIDDEN] = flip_col(df[concept.VISIBLE])
```

Comment:
Why do we want create the HIDDEN col if its not present? We only seem to be using VISIBLE in the kids_first.py target api config..

Reply:
Well I guess if we had both HIDDEN and VISIBLE then i see why, but see my comment above about mirroring..

Reply:
Because the target service mapping only allows for a direct map and while _ours_ uses visible, another might not.